.github/workflows/clang-tidy.yml

```diff
@@ -0,0 +1,51 @@
+name: tidy
+
+on:
+  push:
+    branches:
+      - 'main'
+      - 'aspect-*'
+  pull_request:
+
+concurrency:
+  group: ${{ github.actor }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
```

Comment:
These environment variables are only needed if you want to run ASPECT inside the container. So maybe rename the job to `clang` and also compile and run the unit tests?